modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/MiniPlayer.kt

```diff
@@ -195,6 +219,93 @@ class MiniPlayer @JvmOverloads constructor(context: Context, attrs: AttributeSet
         button.contentDescription = if (isPlaying) stringPause else stringPlay
     }
 
+    private fun loadArtwork(podcast: Podcast?, episode: BaseEpisode) {
+        val imageLoader = PodcastImageLoaderThemed(context)
+        val imageView = binding.artwork
+        imageLoader.radiusPx = 2.dpToPx(context.resources.displayMetrics)
+
+        val artwork = getEpisodeArtwork(episode)
+        if (artwork == Artwork.None && podcast?.uuid != null) {
+            loadPodcastArtwork(imageLoader, podcast)
+        } else {
+            loadEpisodeArtwork(artwork, imageView, imageLoader)?.let { disposable ->
+                launch {
+                    // If episode artwork fails to load, then load podcast artwork
+                    val result = disposable.job.await()
+                    if (result is ErrorResult && podcast?.uuid != null) {
+                        loadPodcastArtwork(imageLoader, podcast)
+                    }
+                }
+            }
+        }
+    }
+
+    private var loadedPodcastUuid: String? = null
+    private fun loadPodcastArtwork(
+        imageLoader: PodcastImageLoaderThemed,
+        podcast: Podcast,
+    ) {
+        if (loadedPodcastUuid == podcast.uuid) return
+        val imageView = binding.artwork
+        imageLoader.smallPlaceholder().loadPodcastUuid(podcast.uuid).into(imageView)
+        loadedPodcastUuid = podcast.uuid
+        loadedEpisodeArtwork = null
+    }
+
+    private var loadedEpisodeArtwork: Artwork? = null
+    private fun loadEpisodeArtwork(
+        artwork: Artwork,
+        imageView: ImageView,
+        imageLoader: PodcastImageLoaderThemed
+    ): Disposable? {
+        if (artwork is Artwork.None || loadedEpisodeArtwork == artwork) return null
+        imageView.imageTintList = null
+
+        val imageBuilder: ImageRequest.Builder.() -> Unit = {
+            error(au.com.shiftyjelly.pocketcasts.images.R.drawable.defaultartwork_dark)
+            size(Size.ORIGINAL)
```

Comment:
Do we need an original size here? We could scale the image to avoid any crash due to large bitmap OOM errors. 

I also think we can extract common methods to load image artwork here and in `PlayerHeaderFragment`. I ask this because we recently fixed this https://github.com/Automattic/pocket-casts-android/pull/1665 for `PlayerHeaderFragment`. If we have a common reusable code, the fix could be applied in both places.

Reply:
Thanks for your review! I've resolved the OOM issue. Extracting the common methods involves tight coupling to fragments. To keep this branch code short and clean, It would be great to keep this as side issue. I would to work upon it. Thanks in advance.